Helidon MP Quickstart 3.2.0 with Jackson fails with Native Image

[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* Helidon Version: 3.2.0
* Helidon MP
* JDK version: GraalVM 22.3.1
* OS: Macos/Linux
* Docker version (if applicable):

Go to Helidon Starter, create Helidon MP Quickstart with *Jackson*, run `mvn clean package -Pnative-image`


```
Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing io.smallrye.openapi.runtime.io.schema.SchemaFactory.lambda$parseSchemaAttr$5(org.eclipse.microprofile.openapi.models.media.Schema$SchemaType, java.lang.Object) 
Parsing context:
   at io.smallrye.openapi.runtime.io.schema.SchemaFactory.lambda$parseSchemaAttr$5(SchemaFactory.java:310)
   at io.smallrye.openapi.runtime.io.schema.SchemaFactory$$Lambda$2982/0x000000030183f910.apply(Unknown Source)
   at io.helidon.config.ConfigMapperManager$MapperProviders.lambda$findMapper$0(ConfigMapperManager.java:191)
   at io.helidon.config.ConfigMapperManager$MapperProviders$$Lambda$1606/0x000000030169d678.apply(Unknown Source)
   at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
   at java.util.TreeMap$KeySpliterator.tryAdvance(TreeMap.java:3088)
   at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
   at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
   at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
   at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
   at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:960)
   at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:934)
   at java.util.stream.AbstractTask.compute(AbstractTask.java:327)
   at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
   at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
   at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:686)
   at java.util.stream.FindOps$FindOp.evaluateParallel(FindOps.java:160)
   at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
   at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
   at java.util.logging.Level$KnownLevel.findByLocalizedLevelName(Level.java:684)
   at java.util.logging.Level.findLevel(Level.java:407)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1011)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)
   at java.util.logging.LogManager$4.run(LogManager.java:971)
   at com.oracle.svm.core.jdk.Target_java_security_AccessController.executePrivileged(SecuritySubstitutions.java:168)
   at java.security.AccessController.doPrivileged(AccessController.java:318)
   at java.lang.Class.newInstance(Class.java:629)
   at java.util.logging.LogManager.createLoggerHandlers(LogManager.java:1005)
   at java.util.logging.LogManager$4.run(LogManager.java:975)

	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisError.parsingError(AnalysisError.java:153)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:104)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:83)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraph(MethodTypeFlow.java:65)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.typestate.DefaultStaticInvokeTypeFlow.update(DefaultStaticInvokeTypeFlow.java:64)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis$1.run(PointsToAnalysis.java:488)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:193)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:177)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of com.fasterxml.jackson.databind.ObjectMapper are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=com.fasterxml.jackson.databind.ObjectMapper.
	at parsing io.smallrye.openapi.runtime.io.JsonUtil.parseValue(JsonUtil.java:286)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2518)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:110)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3393)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3345)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3190)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1138)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1030)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:97)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:84)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:446)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:135)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:685)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.lookupEncodedGraph(InlineBeforeAnalysis.java:180)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.doInline(PEGraphDecoder.java:1162)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.tryInline(PEGraphDecoder.java:1145)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.trySimplifyInvoke(PEGraphDecoder.java:1003)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.handleInvoke(PEGraphDecoder.java:957)
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.processNextNode(GraphDecoder.java:817)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.processNextNode(InlineBeforeAnalysis.java:240)
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.decode(GraphDecoder.java:548)
	at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.decode(PEGraphDecoder.java:833)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysis.decodeGraph(InlineBeforeAnalysis.java:98)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:179)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:349)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:93)
	... 12 more
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of com.fasterxml.jackson.databind.ObjectMapper are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=com.fasterxml.jackson.databind.ObjectMapper.
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:132)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:595)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:177)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:148)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:100)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:79)
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.util.ConstantFoldUtil$1.readValue(ConstantFoldUtil.java:55)
	at jdk.internal.vm.compiler/org.graalvm.compiler.core.common.spi.JavaConstantFieldProvider.readConstantField(JavaConstantFieldProvider.java:78)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantFieldProvider.readConstantField(AnalysisConstantFieldProvider.java:72)
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.util.ConstantFoldUtil.tryConstantFold(ConstantFoldUtil.java:51)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.ConstantFoldLoadFieldPlugin.tryConstantFold(ConstantFoldLoadFieldPlugin.java:53)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.ConstantFoldLoadFieldPlugin.handleLoadStaticField(ConstantFoldLoadFieldPlugin.java:49)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genGetStatic(BytecodeParser.java:4801)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genGetStatic(BytecodeParser.java:4772)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5282)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3385)
	... 37 more
------------------------------------------------------------------------------------------------------------------------
                        0.8s (5.3% of total time) in 20 GCs | Peak RSS: 3.44GB | CPU load: 5.99
========================================================================================================================
Failed generating 'myproject' after 14.7s.
Error: Image build request failed with exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  20.625 s
[INFO] Finished at: 2023-03-31T14:42:16+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.graalvm.buildtools:native-maven-plugin:0.9.16:compile (build-native-image) on project myproject: Execution of /Library/Java/JavaVirtualMachines/graalvm/Contents/Home/bin/native-image @target/tmp/native-image-17100828645439060033.args returned non-zero result -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/integrations/graal/native-image-extension/src/main/resources/META-INF/native-image/io.helidon.integrations.graal/helidon-graal-native-image-extension/native-image.properties
+++ b/integrations/graal/native-image-extension/src/main/resources/META-INF/native-image/io.helidon.integrations.graal/helidon-graal-native-image-extension/native-image.properties
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2019, 2022 Oracle and/or its affiliates.
+# Copyright (c) 2019, 2023 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -22,6 +22,7 @@ Args=--no-fallback \
       --initialize-at-build-time=org.yaml.snakeyaml \
       --initialize-at-build-time=org.reactivestreams \
       --initialize-at-build-time=org.glassfish.json \
+      --initialize-at-build-time=com.fasterxml.jackson \
       --initialize-at-build-time=org.eclipse.microprofile \
       --initialize-at-build-time=io.opentracing \
       --initialize-at-build-time=jakarta.json \

--- a/integrations/graal/native-image-extension/src/main/resources/META-INF/native-image/io.helidon.integrations.graal/helidon-graal-native-image-extension/reflect-config.json
+++ b/integrations/graal/native-image-extension/src/main/resources/META-INF/native-image/io.helidon.integrations.graal/helidon-graal-native-image-extension/reflect-config.json
@@ -13,5 +13,10 @@
         "name": "java.util.logging.FileHandler",
         "allDeclaredConstructors": true,
         "allPublicMethods": true
+    },
+    {
+      "name": "com.fasterxml.jackson.module.jakarta.xmlbind.JakartaXmlBindAnnotationIntrospector",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true
     }
 ]

